pkg/telemetry/builder.go

```diff
@@ -0,0 +1,212 @@
+func (c *Builder) ForgeTelemetryItem(ctx context.Context) (*Telemetry, error) {
+	clusterIdentity, err := utils.GetClusterIdentityWithControllerClient(ctx, c.Client, c.Namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Telemetry{
+		IsInstallation:    c.IsInstallation,
+		ClusterID:         clusterIdentity.ClusterID,
+		LiqoVersion:       c.LiqoVersion,
+		KubernetesVersion: c.KubernetesVersion,
+		Provider:          c.getProvider(),
+		PeeringInfo:       c.getPeeringInfoSlice(ctx),
+		NamespacesInfo:    c.getNamespacesInfo(ctx),
+	}, nil
+}
+
+func (c *Builder) getProvider() string {
+	if len(c.ClusterLabels) == 0 {
+		return ""
+	}
+
+	v, ok := c.ClusterLabels[consts.ProviderClusterLabel]
+	if !ok {
+		return ""
+	}
+	return v
+}
+
+func (c *Builder) getNamespacesInfo(ctx context.Context) []NamespaceInfo {
+	var namespaceOffloadings offloadingv1alpha1.NamespaceOffloadingList
+	if err := c.Client.List(ctx, &namespaceOffloadings); err != nil {
+		// do not fail on errors, but just send other telemetry data
+		klog.Errorf("unable to list namespace offloadings: %v", err)
+		return nil
+	}
+
+	namespaceInfoSlice := make([]NamespaceInfo, len(namespaceOffloadings.Items))
+	for i := range namespaceOffloadings.Items {
+		namespaceOffloading := &namespaceOffloadings.Items[i]
+		namespaceInfoSlice[i] = c.getNamespaceInfo(ctx, namespaceOffloading)
+	}
+	return namespaceInfoSlice
+}
+
+func (c *Builder) getNamespaceInfo(ctx context.Context,
+	namespaceOffloading *offloadingv1alpha1.NamespaceOffloading) NamespaceInfo {
+	namespaceInfo := NamespaceInfo{
+		UID:                string(namespaceOffloading.GetUID()),
+		MappingStrategy:    namespaceOffloading.Spec.NamespaceMappingStrategy,
+		OffloadingStrategy: namespaceOffloading.Spec.PodOffloadingStrategy,
+		HasClusterSelector: len(namespaceOffloading.Spec.ClusterSelector.NodeSelectorTerms) > 0,
+		NumOffloadedPods:   map[string]int64{},
+	}
+
+	offloadedPods, err := liqogetters.ListOffloadedPods(ctx, c.Client, namespaceOffloading.Namespace)
+	runtime.Must(err)
+
+	var nodePodBucket = make(map[string]int64)
+	for i := range offloadedPods.Items {
+		pod := &offloadedPods.Items[i]
+		if pod.Spec.NodeName == "" {
+			klog.Warningf("Pod %s/%s has no node assigned", pod.Namespace, pod.Name)
+			continue
+		}
+
+		if _, ok := nodePodBucket[pod.Spec.NodeName]; !ok {
+			nodePodBucket[pod.Spec.NodeName] = 1
+		} else {
+			nodePodBucket[pod.Spec.NodeName]++
+		}
+	}
+
+	virtualNodes, err := liqogetters.ListVirtualNodes(ctx, c.Client)
+	runtime.Must(err)
+
+	for i := range virtualNodes.Items {
+		node := &virtualNodes.Items[i]
+		clusterID, err := liqogetters.RetrieveClusterIDFromVirtualNode(node)
+		if err != nil {
+			klog.Errorf("unable to get cluster id from node %s: %v", node.Name, err)
+			continue
+		}
+		if _, ok := nodePodBucket[node.Name]; !ok {
+			namespaceInfo.NumOffloadedPods[clusterID] = 0
+			continue
+		}
+		namespaceInfo.NumOffloadedPods[clusterID] = nodePodBucket[node.Name]
```

Comment:
```suggestion
		namespaceInfo.NumOffloadedPods[clusterID] = nodePodBucket[node.Name]
```

Accessing a non existing key returns the default value (0 in this case).